assets/javascripts/discourse/widgets/question-answer-user-commented-notification-item.js

```diff
@@ -0,0 +1,21 @@
+import { DefaultNotificationItem } from "discourse/widgets/default-notification-item";
+import { createWidgetFrom } from "discourse/widgets/widget";
+import { iconNode } from "discourse-common/lib/icon-library";
+import { postUrl } from "discourse/lib/utilities";
+import { buildAnchorId } from "../widgets/qa-comment";
+
+createWidgetFrom(
+  DefaultNotificationItem,
+  "question-answer-user-commented-notification-item",
+  {
+    url(data) {
+      const attrs = this.attrs;
+      const url = postUrl(attrs.slug, attrs.topic_id, attrs.post_number);
+      return `${url}#${buildAnchorId(data.qa_comment_id)}`;
+    },
+
+    icon() {
+      return iconNode("comment");
```

Comment:
This clashes with chat but I'm opened to suggestions about alter icons.

Reply:
A comment is pretty much a reply to an answer, IMO we should use the "reply" icon here. (IN fact I wonder whether the notification type should be `replied`, but that might have other side effects, for example, it might make it harder to pass the fragment to the URL.)

Reply:
> but that might have other side effects,

From previous attempts are trying to use the `Post` model as comments, I've learnt that we should as much as possible avoid coupling comments to the `Post` model. There are so many unintended side effects that are hard to iron out.

Reply:
I've decided to just ship this as it is first since any changes required here are going to be easy to make.

Reply:
Sounds good!